azkaban-web-server/src/main/java/cloudflow/daos/ProjectDaoImpl.java

```diff
@@ -0,0 +1,164 @@
+package cloudflow.daos;
+
+import azkaban.db.DatabaseOperator;
+import azkaban.db.SQLTransaction;
+import azkaban.project.ProjectManagerException;
+import azkaban.user.User;
+import cloudflow.models.Project;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class ProjectDaoImpl implements ProjectDao {
+
+  private DatabaseOperator dbOperator;
+  private ProjectAdminDaoImpl projectAdminDaoImpl;
+  private static final Logger log = LoggerFactory.getLogger(ProjectDaoImpl.class);
+
+  static String INSERT_PROJECT =
+      "INSERT INTO project (name, description, space_id, created_by, creation_time, "
+          + "last_modified_by, last_modified_time, version,  active, enc_type, settings_blob) "
+          + "values "
+          + "(?,?,?,?,?,?,?,?,?,?,?)";
+
+  @Inject
+  public ProjectDaoImpl(DatabaseOperator dbOperator, ProjectAdminDaoImpl projectAdminDaoImpl) {
+    this.dbOperator = dbOperator;
+    this.projectAdminDaoImpl = projectAdminDaoImpl;
+  }
+
+  @Override
+  public String create(Project project, User user) {
+    String name = project.getName();
+    FetchProjectHandler fetchProjectHandler = new FetchProjectHandler();
+
+    // Check if the same project name exists.
+    try {
+      final List<Project> projects = this.dbOperator
+          .query(FetchProjectHandler.SELECT_PROJECT_WITH_NAME_AND_SPACE_ID, fetchProjectHandler,
+              name, project.getSpaceId());
+      if (!projects.isEmpty()) {
+        throw new ProjectManagerException(
+            "Project with name " + name + " already exists.");
+      }
+    } catch (final SQLException ex) {
+      log.error("Check if project '" + name + "' exists failed with: " + ex);
+      throw new ProjectManagerException("Check if project '" + name + "' exists failed");
+    }
+
+    final SQLTransaction<Long> insertAndGetProjectId = transOperator -> {
+      String currentTime = DateTime.now().toLocalDateTime().toString();
```

Comment:
After the POC we shouldn't store dates and times in local time.

Reply:
Created https://docs.google.com/document/d/1S6Jrmi0t8F0Qu8Kqa2MiKwGNnhkwwcvIl1fAb5hCnT4 to track all CloudFlow TODOs. Added this to the list.